can_validator/MY17_Can_Library_h.py

```diff
@@ -0,0 +1,89 @@
+"""
+Generate MY17_Can_Libary.h file.
+Run this file to write just MY17_Can_Libary.h or main.py to write all files.
+"""
+import sys
+sys.path.append("ParseCAN")
+import ParseCAN
+from common import can_lib_h_path, spec_path, struct_paths, unused_messages
+
+
+def write(output_path, spec_path, struct_paths, unused_messages):
+    """
+    Generate MY17_Can_Libary.h file.
+
+    :param output_path: file to be written to
+    :param spec_path: CAN spec path
+    :param struct_paths: list of paths to struct header files (the ones generated by structs.py)
+    :param unused_messages: list of messages that are in the CAN spec but unused
+    """
+    spec = ParseCAN.spec.can(spec_path)
+    with open(output_path, 'w') as f:
+        # Setup file
+        f.write("#ifndef _MY17_CAN_LIBRARY_H\n" +
+                "#define _MY17_CAN_LIBRARY_H\n\n" +
+                "#include <stdint.h>\n" +
+                "#include <stdbool.h>\n\n")
+
+        # Create enum
+        f.write("typedef enum {\n" +
+                "  Can_No_Msg,\n" +
+                "  Can_Unknown_Msg,\n" +
+                "  Can_Error_Msg,\n")
+        for message in spec.messages.values():
+            f.write("  Can_" + message.name + "_Msg,\n")
+
+        # Add motor controller special cases
+        f.write(
+            "  Can_MC_ErrorAndWarning_Msg,\n" +
+            "  Can_MC_DataReading_Msg,\n" +
+            "  Can_MC_State_Msg,\n" +
+            "  Can_Vcu_MCRequest_Msg,\n" +
+            "  Can_Vcu_MCTorque_Msg,\n")
+        f.write("} Can_MsgID_T;\n\n")
+
+        # Finish initial setup
+        f.write(
+            'Can_MsgID_T Can_MsgType(void);\n\n'
```

Comment:
I admire your dedication to 100% generated code, but you should probably just have put these in a separate header file and just `#include`'d it here instead. Still, not worth changing now.

Reply:
👍 to this for MY18